fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/MetaDataProtoEditor.java

```diff
@@ -193,14 +385,200 @@ public static void deprecateRecordType(@Nonnull RecordMetaDataProto.MetaData.Bui
         }
     }
 
+    /**
+     * Rename a record type. This can be used to update any top-level record type defined within the
+     * meta-data's records descriptor, including {@code NESTED} records or the union descriptor. However,
+     * it cannot be used to rename nested messages (i.e., messages defined within other messages) or
+     * records defined in imported files. In addition to updating the file descriptor, if the record type
+     * is not {@code NESTED} or the union descriptor, update any other references to the record type
+     * within the meta-data (such as within index definitions).
+     *
+     * @param metaDataBuilder the meta-data builder
+     * @param recordTypeName the name of the existing top-level record type
+     * @param newRecordTypeName the new name to give to the record type
+     */
+    public static void renameRecordType(@Nonnull RecordMetaDataProto.MetaData.Builder metaDataBuilder,
+                                        @Nonnull String recordTypeName, @Nonnull String newRecordTypeName) {
+        // Create a copy of the records builder (rather than calling metaDataBuilder.getRecordsBuilder()) to avoid
+        // corrupting the records builder in metaDataBuilder before all validation has been done.
+        final DescriptorProtos.FileDescriptorProto records = metaDataBuilder.getRecords();
+        boolean found = false;
+        for (DescriptorProtos.DescriptorProto messageType : records.getMessageTypeList()) {
+            if (messageType.getName().equals(recordTypeName)) {
+                found = true;
+            } else if (messageType.getName().equals(newRecordTypeName)) {
+                throw new MetaDataException("Cannot rename record type to " + newRecordTypeName + " as it already exists");
+            }
+            validateTypeNotInNested(messageType, newRecordTypeName);
+        }
+        if (!found) {
+            throw new MetaDataException("No record type found with name " + recordTypeName);
+        }
+        if (recordTypeName.equals(newRecordTypeName)) {
+            // Identity transformation requires no work.
+            // From here on in, we can assume that recordTypeName != newRecordTypeName, which simplifies things.
+            return;
+        }
+        final DescriptorProtos.FileDescriptorProto.Builder recordsBuilder = records.toBuilder();
+
+        // Determine the usage of the original record type by looking through the union builder.
```

Comment:
You have a comment deep in the nested `if` blocks, but maybe update this comment that the field's name may also change? or, even better, set the name after the top `if` block.

Reply:
Sure, it looks much better now. thanks.